client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -176,6 +180,34 @@ def process(self, instance):
         self.log.debug("Extracted instance '%s' to: %s",
                        instance.name, json_representation)
 
+    def process_container(self, containers):
+        """Allow to collect the asset containers through sub-assembly workflow if
+        there is a layout container
+
+        Args:
+            containers (list): Ayon asset containers
+
+        Returns:
+            dict: container dict with the container and its related asset name
+        """
+        all_containers_set = {}
+        for container in containers:
+            if cmds.getAttr(f"{container}.loader") == "LayoutLoader":
+                # Flatten container layout loader products to their individual loaded containers
+                member_containers = get_container_members(container)
+                # Recursively process each member container
+                children_containers = self.process_container(member_containers)
+                all_containers_set.update(children_containers)
+            else:
+                members = get_container_members(container)
+                transforms = cmds.ls(members, transforms=True)
+                roots = get_highest_in_hierarchy(transforms)
+                root = next(iter(roots), None)
+                if root is not None:
+                    all_containers_set[container] = root
```

Comment:
I wonder, is this more readable?
```suggestion
                if roots:
                    # Assume just a single root node
                    all_containers_set[container] = roots[0]
```